tee-worker/omni-executor/intent/executors/cross-chain/src/lib.rs

```diff
@@ -504,27 +579,386 @@ impl<
+					};
+
+					let remote_signer = RemoteSigner::new(
+						self.pumpx_signer_client.clone(),
+						pumpx_config.wallet_index,
+						*account_id.as_ref(),
+						Handle::current(),
+					);
+
+					// TODO: change this when adding support for more tokens/chains
+					if binance_coin_name == "SOL" {
+						// Native transfer
+						debug!("Transfering {:?} SOL to {:?}", amount_to_transfer, deposit_address);
+						self.solana_client
+							.transfer_sol(&deposit_address, amount_to_transfer, &remote_signer)
+							.await
+							.map_err(|_| {
+								log::error!("Failed to transfer SOL");
+							})?;
+					} else {
+						debug!(
+							"Transfering {:?} {:?} to {:?}",
+							amount_to_transfer, token_address, deposit_address
+						);
+						// SPL transfer
+						self.solana_client
+							.transfer_spl(
+								&deposit_address,
+								amount_to_transfer,
+								&token_address,
+								&remote_signer,
+							)
+							.await
+							.map_err(|_| {
+								log::error!("Failed to transfer SPL");
+							})?;
+					}
+					let (trade_symbol, order_side) = match binance_coin_name.as_str() {
+						"USDC" => ("BNBUSDC".to_string(), BinanceOrderSide::BUY),
+						"USDT" => ("BNBUSDT".to_string(), BinanceOrderSide::BUY),
+						"SOL" => ("SOLBNB".to_string(), BinanceOrderSide::SELL),
+						_ => {
+							log::error!("Unsupported asset: {:?}", binance_coin_name);
+							return Err(());
+						},
+					};
+
+					// 3. Make the trade using binance spot trading api from_asset => BNB, If it fails, notify the backend via /v3/trade/cross_fail
+					let binance_order_params = BinanceCreateOrderParams {
+						symbol: trade_symbol.clone(),
+						side: order_side,
+						order_type: BinanceOrderType::MARKET,
+						..Default::default()
+					};
+					debug!("Creating binance order with params: {:?}", binance_order_params);
+					let Ok(binance_order) =
+						self.binance_api.spot_trading().create_order(binance_order_params).await
+					else {
+						log::error!("Failed to create binance order");
+						let data = CrossOrderFailData {
+							request_id: intent_id,
+							// TODO: is this a user facing error? what should we return?
+							fail_reason: "Failed to create binance order".to_string(),
+						};
+						self.pumpx_api.cross_fail(&access_token, data).await.map_err(|_| {
+							log::error!("Failed to notify pumpx-signer");
+						})?;
+						// TODO: Figure out how to transfer back the asset to the omni account
+						// check https://developers.binance.com/docs/wallet/capital/withdraw
+
+						return Err(());
+					};
+
+					let mut trade_success = false;
+					let mut to_amount = 0;
+					loop {
+						let trade_order = self
+							.binance_api
+							.spot_trading()
+							.get_order(&trade_symbol, Some(binance_order.order_id), None, None)
+							.await
+							.map_err(|_| {
+								log::error!("Failed to get binance order");
+							})?;
+
+						match trade_order.status {
+							BinanceOrderStatus::FILLED => {
```

Comment:
So it means it's fully "filled", right? No partial trade could happen

Reply:
Now I assume we will get FILLED status with all fills (100% completion)

```
	PARTIALLY_FILLED,
	FILLED,
```

But how it behaves is not clear.

Reply:
I see - will we reach it eventually? Like can it be an endless loop?